hnn_qt5.py

```diff
@@ -218,71 +231,174 @@ def runsim (self):
     print(''); self.updatewaitsimwin('')
   
   def optmodel (self):
+    import simdat
+    simdat.initial_ddat = simdat.ddat.copy()
+
     self.updatewaitsimwin('Optimizing model. . .')
-    from neuron import h # for praxis
+
+    self.last_step = False
+    self.first_step = True
+    for step in dconf['opt_info'].keys():
+      self.opt_params = dconf['opt_info'][step]
+
+      self.cur_step = step
+      if self.cur_step == len(dconf['opt_info']) - 1:
+        self.last_step = True
+        # update ranges and recalculate weights
+        self.updateoptparams()
```

Comment:
For the last step (all inputs), recalculate ranges and update dconf['opt_info']. If previous optimization steps increased std. dev. this could result in fewer optimization steps as inputs may be deemed too close together and be grouped in a single optimization step.

The purpose of the last step (with regular RMSE) is to clean up biases introduced by a local weighted RMSE optimizations.